### ISSUE TITLE
Type Error when with Apply Rules To Existing Collection

### ISSUE THREAD

**Issue by assuntad23** [galaxyproject/galaxy#11439]
When running this tutorial: https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-rules-advanced/tutorial.html#apply-rules-to-existing-collections

In the Apply Rules to Existing Collection step, after attempting to execute tool, a server error is displayed and a type error can be found in the console.

Server error: The server could not complete the request. Please contact the Galaxy Team if this error persists. Error executing tool with id '__APPLY_RULES__': Problem applying regular expression, wrong number of groups found.
[followed by JSON]

Console error (appears before clicking Execute, upon Saving the RuleBuilder:
```
instrument.js:110 TypeError: Cannot read property '0' of undefined
    at Object.display (rule-definitions.js:244)
    at a.title (RulesDisplay.vue:51)
    at dn.get (vue.runtime.esm.js:4479)
    at dn.evaluate (vue.runtime.esm.js:4584)
    at a.title (vue.runtime.esm.js:4836)
    at a.render (RulesDisplay.vue:25)
    at a.t._render (vue.runtime.esm.js:3548)
    at a.r (vue.runtime.esm.js:4066)
    at dn.get (vue.runtime.esm.js:4479)
    at new dn (vue.runtime.esm.js:4468)
    at t (vue.runtime.esm.js:4073)
    at a.An.$mount (vue.runtime.esm.js:8415)
    at init (vue.runtime.esm.js:3118)
    at vue.runtime.esm.js:5978
    at f (vue.runtime.esm.js:5925)
    at vue.runtime.esm.js:6216
    at A (vue.runtime.esm.js:6319)
    at vue.runtime.esm.js:6193
    at A (vue.runtime.esm.js:6319)
    at a.__patch__ (vue.runtime.esm.js:6482)
    at a.t._update (vue.runtime.esm.js:3948)
    at a.r (vue.runtime.esm.js:4066)
    at dn.get (vue.runtime.esm.js:4479)
    at dn.run (vue.runtime.esm.js:4554)
    at fn (vue.runtime.esm.js:4310)
    at Array.<anonymous> (vue.runtime.esm.js:1980)
    at Yt (vue.runtime.esm.js:1906)
```

JSON: 
```json
{
    "history_id": "9006f47d54690b05",
    "tool_id": "__APPLY_RULES__",
    "tool_version": "1.1.0",
    "inputs": {
        "input": {
            "values": [
                {
                    "id": "b1ab905a68f00a08",
                    "hid": 37,
                    "name": "anExistingCollection",
                    "src": "hdca",
                    "tags": []
                }
            ],
            "batch": false
        },
        "rules": {
            "rules": [
                {
                    "type": "add_column_metadata",
                    "value": "identifier0",
                    "error": null,
                    "warn": null
                },
                {
                    "type": "add_column_regex",
                    "target_column": 0,
                    "expression": "(.*)_(.*)_.*",
                    "group_count": "2",
                    "error": null,
                    "warn": null
                }
            ],
            "mapping": [
                {
                    "type": "list_identifiers",
                    "columns": [
                        1,
                        2,
                        0
                    ],
                    "editing": false
                }
            ]
        }
    }
}
```

Screenshot (list remains in this state):
![Screenshot from 2021-02-22 15-22-54](https://user-images.githubusercontent.com/26912553/108765505-db62fc00-7521-11eb-9a21-bf8ae3ecbc78.png)

**Comment by jmchilton** [comment]
```
galaxy.tools ERROR 2021-03-02 14:38:12,254 Exception caught while attempting to execute tool with id '__APPLY_RULES__':
Traceback (most recent call last):
  File "/Users/john/workspace/galaxy/lib/galaxy/tools/__init__.py", line 1659, in handle_single_execution
    flush_job=flush_job,
  File "/Users/john/workspace/galaxy/lib/galaxy/tools/__init__.py", line 1747, in execute
    return self.tool_action.execute(self, trans, incoming=incoming, set_output_hid=set_output_hid, history=history, **kwargs)
  File "/Users/john/workspace/galaxy/lib/galaxy/tools/actions/model_operations.py", line 60, in execute
    self._produce_outputs(trans, tool, out_data, output_collections, incoming=incoming, history=history, tags=preserved_tags)
  File "/Users/john/workspace/galaxy/lib/galaxy/tools/actions/model_operations.py", line 76, in _produce_outputs
    tool.produce_outputs(trans, out_data, output_collections, incoming, history=history, tags=tags, tag_handler=tag_handler)
  File "/Users/john/workspace/galaxy/lib/galaxy/tools/__init__.py", line 3185, in produce_outputs
    hdca, rule_set, copy_dataset
  File "/Users/john/workspace/galaxy/lib/galaxy/managers/collections.py", line 492, in apply_rules
    data, sources = rule_set.apply(initial_data, initial_sources)
  File "/Users/john/workspace/galaxy/lib/galaxy/util/rules_dsl.py", line 513, in apply
    data, sources = rule_definition.apply(rule, data, sources)
  File "/Users/john/workspace/galaxy/lib/galaxy/util/rules_dsl.py", line 172, in apply
    return apply_regex(expression, target, data, replacement, group_count), sources
  File "/Users/john/workspace/galaxy/lib/galaxy/util/rules_dsl.py", line 55, in apply_regex
    new_data = list(map(new_row, data))
  File "/Users/john/workspace/galaxy/lib/galaxy/util/rules_dsl.py", line 45, in new_row
    raise Exception("Problem applying regular expression, wrong number of groups found.")
Exception: Problem applying regular expression, wrong number of groups found.
galaxy.tools.execute WARNING 2021-03-02 14:38:12,279 There was a failure executing a job for tool [__APPLY_RULES__] - Error executing tool with id '__APPLY_RULES__': Problem applying regular expression, wrong number of groups found.
```

**Comment by jmchilton** [comment]
Fixed with #11553
